examples/audio_player_example.ex

```diff
@@ -29,68 +29,70 @@ defmodule AudioPlayerConsumer do
   # Audio file will be fed directly to ffmpeg
   @nut_file_url "https://brandthill.com/files/nut.wav"
 
+  # Compile-time helper for defining Discord Application Command options
+  opt = fn type, name, desc, opts ->
+    %{type: type, name: name, description: desc}
+    |> Map.merge(Enum.into(opts, %{}))
+  end
+
+  @play_opts [
+    opt.(1, "song", "Play a song", []),
+    opt.(1, "nut", "Play a nut sound", []),
+    opt.(1, "file", "Play a file", options: [opt.(3, "url", "File URL to play", required: true)]),
+    opt.(1, "url", "Play a URL from a common service",
+      options: [opt.(3, "url", "URL to play", required: true)]
+    )
+  ]
+
+  @commands [
+    {"summon", "Summon bot to your voice channel", []},
+    {"leave", "Tell bot to leave your voice channel", []},
+    {"play", "Play a sound", @play_opts},
+    {"stop", "Stop the playing sound", []},
+    {"pause", "Pause the playing sound", []},
+    {"resume", "Resume the paused sound", []}
+  ]
+
   def start_link do
     Consumer.start_link(__MODULE__)
   end
 
-  def get_voice_channel_of_msg(msg) do
-    msg.guild_id
+  def get_voice_channel_of_interaction(%{guild_id: guild_id, user: %{id: user_id}} = _interaction) do
+    guild_id
     |> GuildCache.get!()
     |> Map.get(:voice_states)
-    |> Enum.find(%{}, fn v -> v.user_id == msg.author.id end)
+    |> Enum.find(%{}, fn v -> v.user_id == user_id end)
     |> Map.get(:channel_id)
   end
 
-  def do_not_ready_msg(msg) do
-    Api.create_message(msg.channel_id, "I need to be in a voice channel for that.")
+  # If you are running this example in an iex session where you manually call
+  # AudioPlayerSupervisor.start_link, you will have to call this function
+  # with your guild_id as the argument
+  def create_guild_commands(guild_id) do
+    Enum.each(@commands, fn {name, description, options} ->
```

Comment:
Would it make sense to run `Enum.map` here, so in case the `create_guild_application_command` fails for whatever reason, it's visible in the return value?